# **Zomato Delivery Data Cleaning**

**Obtenção, Tratamento e Limpeza dos Dados:** Este notebook concentra-se em adquirir, tratar e limpar os dados, garantindo sua qualidade e preparação para as etapas seguintes.

In [1]:
import pandas as pd
import os

# Importando os dados

In [2]:
path = os.getcwd()

In [5]:
df = pd.read_csv(os.path.join(path, 'data/Zomato Dataset.csv'))
df

ID Delivery_person_ID  Delivery_person_Age  \
0      0xcdcd      DEHRES17DEL01                 36.0   
1      0xd987      KOCRES16DEL01                 21.0   
2      0x2784     PUNERES13DEL03                 23.0   
3      0xc8b6     LUDHRES15DEL02                 34.0   
4      0xdb64      KNPRES14DEL02                 24.0   
...       ...                ...                  ...   
45579  0x7c09      JAPRES04DEL01                 30.0   
45580  0xd641      AGRRES16DEL01                 21.0   
45581  0x4f8d     CHENRES08DEL03                 30.0   
45582  0x5eee    COIMBRES11DEL01                 20.0   
45583  0x5fb2   RANCHIRES09DEL02                 23.0   

       Delivery_person_Ratings  Restaurant_latitude  Restaurant_longitude  \
0                          4.2            30.327968             78.046106   
1                          4.7            10.003064             76.307589   
2                          4.7            18.562450             73.916619   
3                          4.3            30.899584             75.809346   
4                          4.7            26.463504             80.372929   
...                        ...                  ...                   ...   
45579                      4.8            26.902328             75.794257   
45580                      4.6             0.000000              0.000000   
45581                      4.9            13.022394             80.242439   
45582                      4.7            11.001753             76.986241   
45583                      4.9            23.351058             85.325731   

       Delivery_location_latitude  Delivery_location_longitude  Order_Date  \
0                       30.397968                    78.116106  12-02-2022   
1                       10.043064                    76.347589  13-02-2022   
2                       18.652450                    74.006619  04-03-2022   
3                       30.919584                    75.829346  13-02-2022   
4                       26.593504                    80.502929  14-02-2022   
...                           ...                          ...         ...   
45579                   26.912328                    75.804257  24-03-2022   
45580                    0.070000                     0.070000  16-02-2022   
45581                   13.052394                    80.272439  11-03-2022   
45582                   11.041753                    77.026241  07-03-2022   
45583                   23.431058                    85.405731  02-03-2022   

      Time_Orderd Time_Order_picked Weather_conditions Road_traffic_density  \
0           21:55             22:10                Fog                  Jam   
1           14:55             15:05             Stormy                 High   
2           17:30             17:40         Sandstorms               Medium   
3           09:20             09:30         Sandstorms                  Low   
4           19:50             20:05                Fog                  Jam   
...           ...               ...                ...                  ...   
45579       11:35             11:45              Windy                 High   
45580       19:55             20:10              Windy                  Jam   
45581       23:50          24:05:00             Cloudy                  Low   
45582       13:35             13:40             Cloudy                 High   
45583       17:10             17:15                Fog               Medium   

       Vehicle_condition Type_of_order Type_of_vehicle  multiple_deliveries  \
0                      2         Snack      motorcycle                  3.0   
1                      1          Meal      motorcycle                  1.0   
2                      1        Drinks         scooter                  1.0   
3                      0        Buffet      motorcycle                  0.0   
4                      1         Snack         scooter                  1.0   
...                  ...           ...   

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45584 entries, 0 to 45583
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   ID                           45584 non-null  object 
 1   Delivery_person_ID           45584 non-null  object 
 2   Delivery_person_Age          43730 non-null  float64
 3   Delivery_person_Ratings      43676 non-null  float64
 4   Restaurant_latitude          45584 non-null  float64
 5   Restaurant_longitude         45584 non-null  float64
 6   Delivery_location_latitude   45584 non-null  float64
 7   Delivery_location_longitude  45584 non-null  float64
 8   Order_Date                   45584 non-null  object 
 9   Time_Orderd                  43853 non-null  object 
 10  Time_Order_picked            45584 non-null  object 
 11  Weather_conditions           44968 non-null  object 
 12  Road_traffic_density         44983 non-null  object 
 13  Vehicle_conditio

# Data Cleansing

### Removendo registro com valores nulos

In [7]:
df[df.isna().any(axis=1)]

ID Delivery_person_ID  Delivery_person_Age  \
8      0x6f67      HYDRES14DEL01                 34.0   
33      0x4f0      MUMRES17DEL02                  NaN   
49     0xb002      JAPRES14DEL01                 32.0   
65      0xa9f     BANGRES13DEL01                  NaN   
77     0x837c     PUNERES14DEL02                 39.0   
...       ...                ...                  ...   
45510  0xd0be      AGRRES07DEL01                  NaN   
45517  0x1ed1      JAPRES18DEL01                  NaN   
45541  0x7483     INDORES04DEL01                 23.0   
45557  0x1163      SURRES06DEL02                  NaN   
45571  0xbb2b      MYSRES08DEL02                 34.0   

       Delivery_person_Ratings  Restaurant_latitude  Restaurant_longitude  \
8                          4.9            17.426228             78.407495   
33                         NaN            19.121999             72.908493   
49                         4.5            26.888420             75.800689   
65                         NaN            12.935662             77.614130   
77                         4.8            18.516216             73.842527   
...                        ...                  ...                   ...   
45510                      NaN            27.160934             78.044095   
45517                      NaN            26.913987             75.752891   
45541                      4.6            22.753659             75.903365   
45557                      NaN            21.185047             72.808590   
45571                      4.9            12.297954             76.665169   

       Delivery_location_latitude  Delivery_location_longitude  Order_Date  \
8                       17.496228                    78.477495  20-03-2022   
33                      19.201999                    72.988493  06-04-2022   
49                      26.898420                    75.810689  26-03-2022   
65                      12.975662                    77.654130  11-03-2022   
77                      18.566216                    73.892527  03-03-2022   
...                           ...                          ...         ...   
45510                   27.170934                    78.054095  13-02-2022   
45517                   27.023987                    75.862891  27-03-2022   
45541                   22.763659                    75.913365  11-03-2022   
45557                   21.205047                    72.828590  03-04-2022   
45571                   12.317954                    76.685169  05-04-2022   

      Time_Orderd Time_Order_picked Weather_conditions Road_traffic_density  \
8           20:40             20:50             Cloudy                  Jam   
33            NaN             18:35             Cloudy               Medium   
49          09:25             09:40              Windy                  Low   
65            NaN             15:30                NaN                  NaN   
77          21:55             22:10             Stormy                  Jam   
...           ...               ...                ...                  ...   
45510         NaN             09:05                NaN                  NaN   
45517       20:35             20:45             Cloudy                  Jam   
45541       08:40             08:45             Cloudy                  Low   
45557         NaN             10:50              Windy                  Low   
45571       09:10             09:15         Sandstorms                  Low   

       Vehicle_condition Type_of_order   Type_of_vehicle  multiple_deliveries  \
8                      0         Snack        motorcycle                  NaN   
33                     1        Drinks           scooter                  1.0   
49                     2        Buffet           scooter                  0.0   
65                     1        Drinks           scooter                  0.0   
77                     1          Meal        motorcycle                  NaN   
...                  ...     

In [8]:
df_clean = df.dropna().copy()
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 41359 entries, 0 to 45583
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   ID                           41359 non-null  object 
 1   Delivery_person_ID           41359 non-null  object 
 2   Delivery_person_Age          41359 non-null  float64
 3   Delivery_person_Ratings      41359 non-null  float64
 4   Restaurant_latitude          41359 non-null  float64
 5   Restaurant_longitude         41359 non-null  float64
 6   Delivery_location_latitude   41359 non-null  float64
 7   Delivery_location_longitude  41359 non-null  float64
 8   Order_Date                   41359 non-null  object 
 9   Time_Orderd                  41359 non-null  object 
 10  Time_Order_picked            41359 non-null  object 
 11  Weather_conditions           41359 non-null  object 
 12  Road_traffic_density         41359 non-null  object 
 13  Vehicle_condition    

### Tratando os registro de latitude e longitude

In [9]:
df_clean['Restaurant_latitude'].value_counts()

Restaurant_latitude
 0.000000     3304
 26.911378     168
 26.914142     167
 26.902328     165
 26.902908     162
              ... 
-27.201725       1
-30.885814       1
-25.451517       1
-30.327968       1
-19.874733       1
Name: count, Length: 487, dtype: int64

* 3304 registros possuem valor de 'Restaurant_latitude' = 0. Vamos remover qualquer registro de localização com valor 0.

In [10]:
df_clean = df_clean[(df_clean['Restaurant_latitude'] != 0) & (df_clean['Restaurant_longitude'] != 0) & (df_clean['Delivery_location_latitude'] != 0) & (df_clean['Delivery_location_longitude'] != 0)]
df_clean

ID Delivery_person_ID  Delivery_person_Age  \
0      0xcdcd      DEHRES17DEL01                 36.0   
1      0xd987      KOCRES16DEL01                 21.0   
2      0x2784     PUNERES13DEL03                 23.0   
3      0xc8b6     LUDHRES15DEL02                 34.0   
4      0xdb64      KNPRES14DEL02                 24.0   
...       ...                ...                  ...   
45578  0x1178   RANCHIRES16DEL01                 35.0   
45579  0x7c09      JAPRES04DEL01                 30.0   
45581  0x4f8d     CHENRES08DEL03                 30.0   
45582  0x5eee    COIMBRES11DEL01                 20.0   
45583  0x5fb2   RANCHIRES09DEL02                 23.0   

       Delivery_person_Ratings  Restaurant_latitude  Restaurant_longitude  \
0                          4.2            30.327968             78.046106   
1                          4.7            10.003064             76.307589   
2                          4.7            18.562450             73.916619   
3                          4.3            30.899584             75.809346   
4                          4.7            26.463504             80.372929   
...                        ...                  ...                   ...   
45578                      4.2            23.371292             85.327872   
45579                      4.8            26.902328             75.794257   
45581                      4.9            13.022394             80.242439   
45582                      4.7            11.001753             76.986241   
45583                      4.9            23.351058             85.325731   

       Delivery_location_latitude  Delivery_location_longitude  Order_Date  \
0                       30.397968                    78.116106  12-02-2022   
1                       10.043064                    76.347589  13-02-2022   
2                       18.652450                    74.006619  04-03-2022   
3                       30.919584                    75.829346  13-02-2022   
4                       26.593504                    80.502929  14-02-2022   
...                           ...                          ...         ...   
45578                   23.481292                    85.437872  08-03-2022   
45579                   26.912328                    75.804257  24-03-2022   
45581                   13.052394                    80.272439  11-03-2022   
45582                   11.041753                    77.026241  07-03-2022   
45583                   23.431058                    85.405731  02-03-2022   

      Time_Orderd Time_Order_picked Weather_conditions Road_traffic_density  \
0           21:55             22:10                Fog                  Jam   
1           14:55             15:05             Stormy                 High   
2           17:30             17:40         Sandstorms               Medium   
3           09:20             09:30         Sandstorms                  Low   
4           19:50             20:05                Fog                  Jam   
...           ...               ...                ...                  ...   
45578       21:45             21:55              Windy                  Jam   
45579       11:35             11:45              Windy                 High   
45581       23:50          24:05:00             Cloudy                  Low   
45582       13:35             13:40             Cloudy                 High   
45583       17:10             17:15                Fog               Medium   

       Vehicle_condition Type_of_order Type_of_vehicle  multiple_deliveries  \
0                      2         Snack      motorcycle                  3.0   
1                      1          Meal      motorcycle                  1.0   
2                      1        Drinks         scooter                  1.0   
3                      0        Buffet      motorcycle                  0.0   
4                      1         Snack         scooter                  1.0   
...                  ...           ...   

In [11]:
df_clean['Restaurant_latitude'].value_counts()

Restaurant_latitude
 26.911378    168
 26.914142    167
 26.902328    165
 26.902908    162
 26.888420    161
             ... 
-30.885814      1
-25.451517      1
-30.327968      1
-26.473698      1
-19.874733      1
Name: count, Length: 486, dtype: int64

### Tratando a coluna 'Order_Date' para datetime

In [12]:
df_clean['Order_Date'].value_counts()

Order_Date
15-03-2022    1000
03-04-2022     990
03-03-2022     990
09-03-2022     988
05-04-2022     982
21-03-2022     981
13-03-2022     979
26-03-2022     974
19-03-2022     970
24-03-2022     970
07-03-2022     969
01-03-2022     969
05-03-2022     968
01-04-2022     967
11-03-2022     960
28-03-2022     959
30-03-2022     953
17-03-2022     946
02-03-2022     864
10-03-2022     854
16-03-2022     853
06-03-2022     850
20-03-2022     849
02-04-2022     848
29-03-2022     845
04-03-2022     842
18-03-2022     837
23-03-2022     836
08-03-2022     836
25-03-2022     834
31-03-2022     833
12-03-2022     830
27-03-2022     825
14-03-2022     821
06-04-2022     813
04-04-2022     805
11-02-2022     737
15-02-2022     708
13-02-2022     704
17-02-2022     703
16-02-2022     666
18-02-2022     654
12-02-2022     648
14-02-2022     645
Name: count, dtype: int64

In [13]:
pd.options.mode.chained_assignment = None
df_clean['Order_Date'] = pd.to_datetime(df_clean['Order_Date'], format='%d-%m-%Y', errors='coerce')
df_clean.head()

ID Delivery_person_ID  Delivery_person_Age  Delivery_person_Ratings  \
0  0xcdcd      DEHRES17DEL01                 36.0                      4.2   
1  0xd987      KOCRES16DEL01                 21.0                      4.7   
2  0x2784     PUNERES13DEL03                 23.0                      4.7   
3  0xc8b6     LUDHRES15DEL02                 34.0                      4.3   
4  0xdb64      KNPRES14DEL02                 24.0                      4.7   

   Restaurant_latitude  Restaurant_longitude  Delivery_location_latitude  \
0            30.327968             78.046106                   30.397968   
1            10.003064             76.307589                   10.043064   
2            18.562450             73.916619                   18.652450   
3            30.899584             75.809346                   30.919584   
4            26.463504             80.372929                   26.593504   

   Delivery_location_longitude Order_Date Time_Orderd Time_Order_picked  \
0                    78.116106 2022-02-12       21:55             22:10   
1                    76.347589 2022-02-13       14:55             15:05   
2                    74.006619 2022-03-04       17:30             17:40   
3                    75.829346 2022-02-13       09:20             09:30   
4                    80.502929 2022-02-14       19:50             20:05   

  Weather_conditions Road_traffic_density  Vehicle_condition Type_of_order  \
0                Fog                  Jam                  2         Snack   
1             Stormy                 High                  1          Meal   
2         Sandstorms               Medium                  1        Drinks   
3         Sandstorms                  Low                  0        Buffet   
4                Fog                  Jam                  1         Snack   

  Type_of_vehicle  multiple_deliveries Festival           City  \
0      motorcycle                  3.0       No  Metropolitian   
1      motorcycle                  1.0       No  Metropolitian   
2         scooter                  1.0       No  Metropolitian   
3      motorcycle                  0.0       No  Metropolitian   
4         scooter                  1.0       No  Metropolitian   

   Time_taken (min)  
0                46  
1                23  
2                21  
3                20  
4                41

In [14]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 38055 entries, 0 to 45583
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   ID                           38055 non-null  object        
 1   Delivery_person_ID           38055 non-null  object        
 2   Delivery_person_Age          38055 non-null  float64       
 3   Delivery_person_Ratings      38055 non-null  float64       
 4   Restaurant_latitude          38055 non-null  float64       
 5   Restaurant_longitude         38055 non-null  float64       
 6   Delivery_location_latitude   38055 non-null  float64       
 7   Delivery_location_longitude  38055 non-null  float64       
 8   Order_Date                   38055 non-null  datetime64[ns]
 9   Time_Orderd                  38055 non-null  object        
 10  Time_Order_picked            38055 non-null  object        
 11  Weather_conditions           38055 non-null  o

### Tratando as colunas de horas

In [15]:
df_clean['Time_Orderd'].value_counts().sort_index()

Time_Orderd
0.375          155
0.416666667    174
0.458333333    162
0.5            168
0.541666667     71
              ... 
23:35          357
23:40          362
23:45          332
23:50          329
23:55          358
Name: count, Length: 176, dtype: int64

In [16]:
df_clean['Time_Order_picked'].value_counts().sort_index()

Time_Order_picked
0.375          173
0.416666667    164
0.458333333    124
0.5            148
0.541666667     65
              ... 
23:50          366
23:55          329
24:05:00       360
24:10:00       228
24:15:00       134
Name: count, Length: 193, dtype: int64

* Vários registros possuem valores diferentes do padrão de hora. Vamos definir um padrão 00:00 e remover os valores diferentes desse formato.

In [17]:
time_pattern = r'^\d{2}:\d{2}$'
df_clean = df_clean[df_clean['Time_Orderd'].str.match(time_pattern)]
df_clean = df_clean[df_clean['Time_Order_picked'].str.match(time_pattern)]
df_clean.head(6)

ID Delivery_person_ID  Delivery_person_Age  Delivery_person_Ratings  \
0  0xcdcd      DEHRES17DEL01                 36.0                      4.2   
1  0xd987      KOCRES16DEL01                 21.0                      4.7   
2  0x2784     PUNERES13DEL03                 23.0                      4.7   
3  0xc8b6     LUDHRES15DEL02                 34.0                      4.3   
4  0xdb64      KNPRES14DEL02                 24.0                      4.7   
5  0x3af3      MUMRES15DEL03                 29.0                      4.5   

   Restaurant_latitude  Restaurant_longitude  Delivery_location_latitude  \
0            30.327968             78.046106                   30.397968   
1            10.003064             76.307589                   10.043064   
2            18.562450             73.916619                   18.652450   
3            30.899584             75.809346                   30.919584   
4            26.463504             80.372929                   26.593504   
5            19.176269             72.836721                   19.266269   

   Delivery_location_longitude Order_Date Time_Orderd Time_Order_picked  \
0                    78.116106 2022-02-12       21:55             22:10   
1                    76.347589 2022-02-13       14:55             15:05   
2                    74.006619 2022-03-04       17:30             17:40   
3                    75.829346 2022-02-13       09:20             09:30   
4                    80.502929 2022-02-14       19:50             20:05   
5                    72.926721 2022-04-02       20:25             20:35   

  Weather_conditions Road_traffic_density  Vehicle_condition Type_of_order  \
0                Fog                  Jam                  2         Snack   
1             Stormy                 High                  1          Meal   
2         Sandstorms               Medium                  1        Drinks   
3         Sandstorms                  Low                  0        Buffet   
4                Fog                  Jam                  1         Snack   
5         Sandstorms                  Jam                  2        Buffet   

    Type_of_vehicle  multiple_deliveries Festival           City  \
0        motorcycle                  3.0       No  Metropolitian   
1        motorcycle                  1.0       No  Metropolitian   
2           scooter                  1.0       No  Metropolitian   
3        motorcycle                  0.0       No  Metropolitian   
4           scooter                  1.0       No  Metropolitian   
5  electric_scooter                  1.0       No  Metropolitian   

   Time_taken (min)  
0                46  
1                23  
2                21  
3                20  
4                41  
5                20

In [18]:
df_clean['Time_Order_picked'].value_counts().sort_index()

Time_Order_picked
08:15     41
08:20     96
08:25    172
08:30    162
08:35    147
        ... 
23:35    369
23:40    340
23:45    358
23:50    366
23:55    329
Name: count, Length: 174, dtype: int64

### Verificando outras colunas

In [19]:
df_clean['Weather_conditions'].value_counts()

Weather_conditions
Fog           5228
Stormy        5190
Windy         5130
Sandstorms    5122
Cloudy        5102
Sunny         4983
Name: count, dtype: int64

In [20]:
df_clean['Road_traffic_density'].value_counts()

Road_traffic_density
Low       10096
Jam        9966
Medium     7666
High       3027
Name: count, dtype: int64

In [21]:
df_clean['Vehicle_condition'].value_counts()

Vehicle_condition
0    10360
2    10207
1    10188
Name: count, dtype: int64

In [22]:
df_clean['Type_of_order'].value_counts()

Type_of_order
Drinks    7745
Snack     7735
Meal      7722
Buffet    7553
Name: count, dtype: int64

In [23]:
df_clean['Type_of_vehicle'].value_counts()

Type_of_vehicle
motorcycle          18162
scooter             10148
electric_scooter     2445
Name: count, dtype: int64

In [24]:
df_clean['multiple_deliveries'].value_counts()

multiple_deliveries
1.0    19437
0.0     9642
2.0     1425
3.0      251
Name: count, dtype: int64

In [25]:
df_clean['Festival'].value_counts()

Festival
No     30133
Yes      622
Name: count, dtype: int64

In [26]:
df_clean['City'].value_counts()

City
Metropolitian    23704
Urban             6927
Semi-Urban         124
Name: count, dtype: int64

In [27]:
df_clean['Time_taken (min)'].value_counts().sort_index()

Time_taken (min)
10     429
11     441
12     443
13     468
14     453
15    1150
16    1118
17    1127
18    1164
19    1188
20    1091
21    1032
22    1077
23    1131
24    1136
25    1430
26    1467
27    1370
28    1340
29    1340
30     827
31     848
32     774
33     895
34     805
35     570
36     590
37     582
38     634
39     592
40     396
41     391
42     402
43     403
44     389
45     173
46     202
47     200
48     194
49     192
50      49
51      64
52      55
53      68
54      65
Name: count, dtype: int64

# Exportando o arquivo CSV

In [28]:
df_clean

ID Delivery_person_ID  Delivery_person_Age  \
0      0xcdcd      DEHRES17DEL01                 36.0   
1      0xd987      KOCRES16DEL01                 21.0   
2      0x2784     PUNERES13DEL03                 23.0   
3      0xc8b6     LUDHRES15DEL02                 34.0   
4      0xdb64      KNPRES14DEL02                 24.0   
...       ...                ...                  ...   
45576  0xc9ab      KNPRES03DEL01                 30.0   
45578  0x1178   RANCHIRES16DEL01                 35.0   
45579  0x7c09      JAPRES04DEL01                 30.0   
45582  0x5eee    COIMBRES11DEL01                 20.0   
45583  0x5fb2   RANCHIRES09DEL02                 23.0   

       Delivery_person_Ratings  Restaurant_latitude  Restaurant_longitude  \
0                          4.2            30.327968             78.046106   
1                          4.7            10.003064             76.307589   
2                          4.7            18.562450             73.916619   
3                          4.3            30.899584             75.809346   
4                          4.7            26.463504             80.372929   
...                        ...                  ...                   ...   
45576                      4.2            26.469003             80.316344   
45578                      4.2            23.371292             85.327872   
45579                      4.8            26.902328             75.794257   
45582                      4.7            11.001753             76.986241   
45583                      4.9            23.351058             85.325731   

       Delivery_location_latitude  Delivery_location_longitude Order_Date  \
0                       30.397968                    78.116106 2022-02-12   
1                       10.043064                    76.347589 2022-02-13   
2                       18.652450                    74.006619 2022-03-04   
3                       30.919584                    75.829346 2022-02-13   
4                       26.593504                    80.502929 2022-02-14   
...                           ...                          ...        ...   
45576                   26.539003                    80.386344 2022-02-14   
45578                   23.481292                    85.437872 2022-03-08   
45579                   26.912328                    75.804257 2022-03-24   
45582                   11.041753                    77.026241 2022-03-07   
45583                   23.431058                    85.405731 2022-03-02   

      Time_Orderd Time_Order_picked Weather_conditions Road_traffic_density  \
0           21:55             22:10                Fog                  Jam   
1           14:55             15:05             Stormy                 High   
2           17:30             17:40         Sandstorms               Medium   
3           09:20             09:30         Sandstorms                  Low   
4           19:50             20:05                Fog                  Jam   
...           ...               ...                ...                  ...   
45576       18:10             18:25             Cloudy               Medium   
45578       21:45             21:55              Windy                  Jam   
45579       11:35             11:45              Windy                 High   
45582       13:35             13:40             Cloudy                 High   
45583       17:10             17:15                Fog               Medium   

       Vehicle_condition Type_of_order Type_of_vehicle  multiple_deliveries  \
0                      2         Snack      motorcycle                  3.0   
1                      1          Meal      motorcycle                  1.0   
2                      1        Drinks         scooter                  1.0   
3                      0        Buffet      motorcycle                  0.0   
4                      1         Snack         scooter                  1.0   
...                  ...           ...             ..

In [29]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30755 entries, 0 to 45583
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   ID                           30755 non-null  object        
 1   Delivery_person_ID           30755 non-null  object        
 2   Delivery_person_Age          30755 non-null  float64       
 3   Delivery_person_Ratings      30755 non-null  float64       
 4   Restaurant_latitude          30755 non-null  float64       
 5   Restaurant_longitude         30755 non-null  float64       
 6   Delivery_location_latitude   30755 non-null  float64       
 7   Delivery_location_longitude  30755 non-null  float64       
 8   Order_Date                   30755 non-null  datetime64[ns]
 9   Time_Orderd                  30755 non-null  object        
 10  Time_Order_picked            30755 non-null  object        
 11  Weather_conditions           30755 non-null  o

In [30]:
df_clean.to_csv(os.path.join(path, 'data/zomato_clean.csv'), index=False)